src/Core/Checkout/Cart/CartRuleLoader.php

```diff
@@ -220,12 +221,21 @@ private function detectTaxType(SalesChannelContext $context, float $cartNetAmoun
         }
 
         $country = $context->getShippingLocation()->getCountry();
+        $customer = $context->getCustomer();
 
-        $isReachedCustomerTaxFreeAmount = $country->getCustomerTax()->getEnabled() && $this->isReachedCountryTaxFreeAmount($context, $country, $cartNetAmount);
-        $isReachedCompanyTaxFreeAmount = $this->taxDetector->isCompanyTaxFree($context, $country) && $this->isReachedCountryTaxFreeAmount($context, $country, $cartNetAmount, CountryDefinition::TYPE_COMPANY_TAX_FREE);
+        if ($customer?->getAccountType() === CustomerEntity::ACCOUNT_TYPE_BUSINESS) {
```

Comment:
not needed elsewhere
```suggestion
        if ($context->getCustomer()?->getAccountType() === CustomerEntity::ACCOUNT_TYPE_BUSINESS) {
```